File: java_solutions/2490.circular-sentence.java
/*
 * @lc app=leetcode id=2490 lang=java
 *
 * [2490] Circular Sentence
 */

class Solution {
    public boolean isCircularSentence(String s) {
        return s.charAt(0) == s.charAt(s.length() - 1) && 
            IntStream.range(0, s.length())
                .filter(i -> s.charAt(i) == ' ')
                .allMatch(i -> s.charAt(i - 1) == s.charAt(i + 1));
    }
}

File: kotlin_solutions/2490.circular-sentence.kt
/*
 * @lc app=leetcode id=2490 lang=kotlin
 *
 * [2490] Circular Sentence
 */

class Solution {fun isCircularSentence(sentence: String) = sentence.split(" ").let { it.zipWithNext().all { (a, b) -> a.last() == b.first() } && it.first().first() == it.last().last() }}

